File: apps/mobile/android/shared/src/commonMain/kotlin/cc/edecan/shared/ChatSecretRedaction.kt
package cc.edecan.shared

/** Evita pintar secretos en la burbuja optimista antes de que el master los
 * reciba, cifre y persista ya redactados. */
object ChatSecretRedaction {
    private val patterns = listOf(
        Regex("\\bsk[-_][A-Za-z0-9_-]{8,}", RegexOption.IGNORE_CASE),
        Regex("\\bBearer\\s+[A-Za-z0-9._~+/=-]{8,}", RegexOption.IGNORE_CASE),
        Regex("\\b(?:rk_live|rk_test|whsec)_[A-Za-z0-9]{8,}", RegexOption.IGNORE_CASE),
        Regex("\\b(?:AKIA|ASIA)[A-Z0-9]{16}\\b"),
    )

    fun redact(text: String): String = patterns.fold(text) { safe, pattern ->
        pattern.replace(safe, "[credencial protegida]")
    }
}

File: apps/mobile/ios/EdecanKit/Sources/EdecanKit/ChatSecretRedaction.swift
import Foundation

/// Redacción local para que una credencial nunca se pinte en una burbuja
/// optimista mientras el request viaja al Edecán maestro.
public enum ChatSecretRedaction {
    private static let patterns: [NSRegularExpression] = [
        #"\bsk[-_][A-Za-z0-9_-]{8,}"#,
        #"\bBearer\s+[A-Za-z0-9._~+/=-]{8,}"#,
        #"\b(?:rk_live|rk_test|whsec)_[A-Za-z0-9]{8,}"#,
        #"\b(?:AKIA|ASIA)[A-Z0-9]{16}\b"#,
    ].compactMap { try? NSRegularExpression(pattern: $0, options: [.caseInsensitive]) }

    public static func redact(_ text: String) -> String {
        patterns.reduce(text) { current, regex in
            let range = NSRange(current.startIndex..<current.endIndex, in: current)
            return regex.stringByReplacingMatches(
                in: current,
                range: range,
                withTemplate: "[credencial protegida]"
            )
        }
    }
}
